src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -169,6 +174,29 @@ public void handleLoanUpdate(LoanDTO loan) {
     }
   }
 
+  @Override
+  public void handleRequestUpdate(RequestDTO requestDTO) {
+    var transaction = transactionRepository.fetchActiveByRequestId(requestDTO.getId()).orElse(null);
+    if (transaction == null) {
+      return;
+    }
+
+    if (transaction.getType() != ITEM) {
+      log.info("Transaction {} isn't Item Hold", transaction.getId());
+      return;
+    }
+
+    var transactionItemHold = (TransactionItemHold) transaction.getHold();
+    if (!transactionItemHold.getFolioItemId().equals(requestDTO.getItemId())) {
+      log.info("Updating transaction {} on request to transfer {}", transaction.getId(), requestDTO.getId());
+      transactionItemHold.setFolioItemId(requestDTO.getItemId());
```

Comment:
there are multiple fields that potentially need to be updated when the request is moved to another item:
```
  folioInstanceId
  folioHoldingId
  folioItemId
  folioItemBarcode
```

Reply:
Resolved